src/OrleansRuntime/GrainDirectory/MultiClusterRegistration/RegistrarManager.cs

```diff
@@ -11,32 +11,15 @@ namespace Orleans.Runtime.GrainDirectory
     internal class RegistrarManager
     {
         private readonly GrainTypeManager grainTypeManager;
+        private readonly object registrarLock = new object();
+        private readonly IServiceProvider serviceProvider;
 
-        private readonly Dictionary<Type, IGrainRegistrar> registrars = new Dictionary<Type, IGrainRegistrar>();
+        private Dictionary<Type, IGrainRegistrar> registrars = new Dictionary<Type, IGrainRegistrar>();
```

Comment:
I'd use IReadOnlyDictionary as the field type, since the intent is to always clone, update and swap.